pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -298,19 +295,23 @@ func (exp *flowExporter) sendFlowRecords() error {
 				}
 				exp.numDataSetsSent = exp.numDataSetsSent + 1
 			}
-			// If the connection is not present in the conntrack table anymore,
-			// then we delete the connection in connection store and record store.
-			// Please note that our record cache policy is that we keep the record in
-			// the record map as long as the corresponding connection is present in
-			// the conntrack table.
-			if !record.Conn.IsPresent {
-				klog.V(2).Infof("Deleting the inactive connection with key: %v from both connection map and record map", key)
-				if err := exp.connStore.DeleteConnectionByKey(key); err != nil {
+
+			if flowexporter.IsConnectionDying(&record.Conn) {
+				// If the connection is in the dying state, then we will set the flow record
+				// to nil after sending it.
+				if err := exp.flowRecords.SetFlowRecordToNilWithoutLock(key); err != nil {
 					return err
 				}
+			}
+
+			if !record.Conn.IsPresent {
```

Comment:
Deletion of connection in connection map logic changed again?

Reply:
Yes I changed a little bit. Mainly because if we delete records by key for dying connection, when we update records, we will add the records again -> and if we move skipping logic (for dying connection) at the beginning of addOrUpdateRecords, we may lose the records that first coming in dying state. 

Reason for keeping the deletion of isPresent here is for easier unit tests (this part can be put either here or reset connection part)